### PR TITLE
Add tournament style generated maps

### LINKED ISSUE #2004: Add option to not have preview in generated maps
This would be ideal for generating maps for tournaments

### PR DESCRIPTION
**Pull request by Sheikah45**
Fixes #2004

### CODE REVIEW

File: src/main/java/com/faforever/client/game/GenerateMapController.java [hunk 1]
```diff
@@ -179,7 +190,11 @@ private byte getSliderValue(Slider slider, CheckBox checkBox) {
     byte plateauDensity = getSliderValue(plateauSlider, plateauRandom);
     byte mountainDensity = getSliderValue(mountainSlider, mountainRandom);
     byte rampDensity = getSliderValue(rampSlider, rampRandom);
-    return new byte[]{spawnCount, mapSize, landDensity, plateauDensity, mountainDensity, rampDensity};
+    if (!fullRandom.isSelected()) {
+      return new byte[]{spawnCount, mapSize, landDensity, plateauDensity, mountainDensity, rampDensity};
+    } else {
+      return new byte[]{spawnCount, mapSize};
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/game/GenerateMapController.java:214]: What has no preview to do with not being able to set all other options

Comment (Sheikah45) [reply src/main/java/com/faforever/client/game/GenerateMapController.java:214]: I changed the title the idea is to have it be for tournament related things where full random is a given

**Comment by Sheikah45** [comment]
gonna change this

**Comment by Sheikah45** [comment]
Alright should be good now

File: src/main/java/com/faforever/client/map/generator/GenerationType.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+public enum GenerationType {
+  CASUAL("game.generateMap.casual"),
+  TOURNAMENT("game.generateMap.tournament"),
+  BLIND("game.generateMap.blind");
+
+  @Getter
+  private final String i18nKey;
+
+  GenerationType(String i18nKey) {
```
Comment (Brutus5000) [inline src/main/java/com/faforever/client/map/generator/GenerationType.java:13]: @AllArgsConstructor instead?

**Review by Brutus5000 — APPROVED** [review]
I didn't test it but from a code quality perspective it looks good

**Pull request merged**